From yanmin_zhang@linux.intel.com  Thu Jun  7 16:47:40 2007
From: Stephen Hemminger <shemminger@linux-foundation.org>
Date: Wed, 06 Jun 2007 11:46:49 +0800
Subject: PCI aer: fix stub return values
To: gregkh@suse.de
Cc: akpm@linux-foundation.org, michael@ellerman.id.au, shemminger@linux-foundation.org
Message-ID: <1181101609.3068.25.camel@ymzhang>

From: Stephen Hemminger <shemminger@linux-foundation.org>

The stubs used when advanced error reporting is not enabled
must have same return type as real functions.

Signed-off-by: Stephen Hemminger <shemminger@linux-foundation.org>
Acked-by: Zhang Yanmin <yanmin.zhang@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 include/linux/aer.h |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/include/linux/aer.h
+++ b/include/linux/aer.h
@@ -14,10 +14,10 @@ extern int pci_find_aer_capability(struc
 extern int pci_disable_pcie_error_reporting(struct pci_dev *dev);
 extern int pci_cleanup_aer_uncorrect_error_status(struct pci_dev *dev);
 #else
-#define pci_enable_pcie_error_reporting(dev)		do { } while (0)
-#define pci_find_aer_capability(dev)			do { } while (0)
-#define pci_disable_pcie_error_reporting(dev)		do { } while (0)
-#define pci_cleanup_aer_uncorrect_error_status(dev)	do { } while (0)
+#define pci_enable_pcie_error_reporting(dev)		(-EINVAL)
+#define pci_find_aer_capability(dev)			(0)
+#define pci_disable_pcie_error_reporting(dev)		(-EINVAL)
+#define pci_cleanup_aer_uncorrect_error_status(dev)	(-EINVAL)
 #endif
 
 #endif //_AER_H_
